rust/azure_iot_operations_connector/src/filemount/connector_config.rs

```diff
@@ -0,0 +1,335 @@
+            if let Some(ca_trustbundle_path) = &self.broker_ca_cert_trustbundle_path {
+                let mut d = std::fs::read_dir(ca_trustbundle_path)
+                    .map_err(|e| format!("Could not read trustbundle directory: {e}"))?;
+                let entry = d
+                    .next()
+                    .ok_or("No CA cert found in trustbundle directory".to_string())?
+                    .map_err(|e| format!("Could not read trustbundle directory: {e}"))?;
+                if d.next().is_some() {
+                    Err("MQTTConnectionSettings only supports a single CA cert".to_string())?
+                } else {
+                    // Convert filepath to string for MqttConnectionSettings
+                    let path_s = entry
+                        .path()
+                        .to_str()
+                        .ok_or("Could not convert Path to String".to_string())?
+                        .to_string();
+                    Some(path_s)
+                }
+            } else {
+                None
+            }
+        };
+
+        let c = aio_mqtt::MqttConnectionSettingsBuilder::default()
+            .client_id(client_id)
+            .hostname(hostname)
+            .tcp_port(tcp_port)
+            .keep_alive(keep_alive)
+            .receive_max(receive_max)
+            .session_expiry(session_expiry)
+            .use_tls(use_tls)
+            .ca_file(ca_file)
+            .sat_file(sat_file)
+            .build()
+            .map_err(|e| format!("{e}"))?;
+        Ok(c)
+    }
+
+    fn extract_mqtt_connection_configuration(
+        mount_path: &Path,
+    ) -> Result<MqttConnectionConfiguration, DeploymentArtifactErrorRepr> {
+        let mqtt_conn_config_pathbuf = mount_path.join("MQTT_CONNECTION_CONFIGURATION");
+        if !mqtt_conn_config_pathbuf.as_path().exists() {
+            return Err(DeploymentArtifactErrorRepr::FilePathMissing(
+                mqtt_conn_config_pathbuf.into_os_string(),
+            ))?;
+        }
+        // NOTE: Manual file read to memory is more efficient than using serde_json::from_reader()
+        let m: MqttConnectionConfiguration =
+            serde_json::from_str(&std::fs::read_to_string(&mqtt_conn_config_pathbuf)?)?;
+        Ok(m)
+    }
+
+    fn extract_aio_metadata(mount_path: &Path) -> Result<AioMetadata, DeploymentArtifactErrorRepr> {
+        let aio_metadata_pathbuf = mount_path.join("AIO_METADATA");
+        if !aio_metadata_pathbuf.as_path().exists() {
+            return Err(DeploymentArtifactErrorRepr::FilePathMissing(
+                aio_metadata_pathbuf.into_os_string(),
+            ))?;
+        }
+        // NOTE: Manual file read to memory is more efficient than using serde_json::from_reader()
+        let a: AioMetadata =
+            serde_json::from_str(&std::fs::read_to_string(&aio_metadata_pathbuf)?)?;
+        Ok(a)
+    }
+
+    fn extract_diagnostics(mount_path: &Path) -> Result<Diagnostics, DeploymentArtifactErrorRepr> {
```

Comment:
Could this be a trait (along with all extract_*) functions

Let's revisit at some point